src/main/java/com/shopify/ShopifySdk.java

```diff
@@ -913,23 +934,34 @@ private void setVariantImageIds(final ShopifyProductRequest shopifyProductReques
 	}
 
 	private Response get(final WebTarget webTarget) {
-		final Callable<Response> responseCallable = () -> webTarget.request(MediaType.APPLICATION_JSON)
-				.header(ACCESS_TOKEN_HEADER, accessToken).get();
+		final Callable<Response> responseCallable = () -> buildCall(webTarget).get();
 		final Response response = invokeResponseCallable(responseCallable);
 		return handleResponse(response, Status.OK);
 	}
 
+	private Invocation.Builder buildCall(WebTarget webTarget) {
+		Invocation.Builder builder = webTarget.request(MediaType.APPLICATION_JSON);
+		if (null != apiKey && null != password) {
```

Comment:
Thanks for adding this. Can you add or update a unit test to use this flow so we can get some coverage for regression.